### PR TITLE
Add cpp loop version

### PR DESCRIPTION
**Pull request by hare1039**
Pass by and add the std modern c++ version.

### CODE REVIEW

File: loops/cpp/code.cpp
Comment (Hxnter999) [inline loops/cpp/code.cpp:1]: looks good; consider `std::println` though.

File: loops/cpp/code.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    int const mod = std::stoi(argv[1]);                       // Get an input number from the command line
+    std::random_device rd{};
+    std::mt19937 engine{rd()};
+    std::uniform_int_distribution<int> dist{0, 9999};         // <random> library setup
+
+    int const picked_number = dist(engine);                   // Get a random number 0 <= r < 10k
+    std::array<std::int32_t, 10000> array{};                  // Array of 10k elements initialized to 0
+    for (int const i : std::ranges::iota_view{0, 10000})      // 10k outer loop iterations
+    {
+        for (int const j : std::ranges::iota_view{0, 100000}) // 100k inner loop iterations, per outer loop iteration
```
Comment (bddicken) [inline loops/cpp/code.cpp:19]: Idk what iota_view is, can you explain?

Comment (safetycoomer) [reply loops/cpp/code.cpp:19]: a range type, iterable over the provided values https://en.cppreference.com/w/cpp/ranges/iota_view

File: loops/cpp/code.cpp
Comment (hare1039) [reply loops/cpp/code.cpp:1]: Thanks for the suggestion. Updated the code.

File: loops/cpp/code.cpp (hunk 1 above)
Comment (hare1039) [reply loops/cpp/code.cpp:19]: Its similar to python's `range(0, 10000)`

Comment (hare1039) [reply loops/cpp/code.cpp]: Switched to `std::ranges::views::iota` for clarity.

**Comment by bddicken** [comment]
Thanks for contributing!

**Pull request merged**